File: JJB/231124/최빈값_구하기.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<int> array) 
{
    int answer = 0;
    
    unordered_map<int, int> mode;
    for(const auto &num : array)
        mode[num]++;
    
    int max = 0;
    for(const auto &data : mode)
    {
        if(data.second > max)
        {
            max = data.second;
            answer = data.first;
        }
        else if(data.second == max)
            answer = -1;
    }
    
    return answer;
}

File: JJB/Programmers/231124/최빈값_구하기.py
def solution(array):
    answer = 0
    
    array_dic = {}
    for i in array:
        try:
            array_dic[i] += 1
        except:
            array_dic[i] = 1
            
    mode = -1
    for key, value in array_dic.items():
        if value > mode:
            mode = value
            answer = key
        elif value == mode:
            answer = -1
        
    return answer